src/disrupt.sh

```diff
@@ -1,15 +1,35 @@
 #!/bin/bash
 
 # Export all these variables for SC2155 rule
-export facthtml
-export tmpstring
+export RANDOM_FACT_HTML
+export BASE_64_IMAGE
 
-randomfacts="$(shuf -n 1 "$FACTS_SOURCE")"
-facthtml="$(echo "${randomfacts}" | sed 's/&/\&amp;/g; s/</\&lt;/g; s/>/\&gt;/g; s/"/\&quot;/g; s/'"'"'/\&#39;/g')"
-tmpstring="$(eval "echo \"$(cat "$RUN_DIR"/assets/template)\"")"
+RANDOM_FACT="$(shuf -n 1 "${FACTS_SOURCE}")"
+RANDOM_FACT_HTML="$(echo "${RANDOM_FACT}" | sed 's/&/\&amp;/g; s/</\&lt;/g; s/>/\&gt;/g; s/"/\&quot;/g; s/'"'"'/\&#39;/g')"
+
+DICE_ROLL=$(shuf -i 1-"${IMAGE_RARITY}" -n 1)
 
 if [ "$#" -eq 1 ]; then
-    echo "$tmpstring" | /usr/bin/mutt -e 'set content_type=text/html' -s "lucashi fact of the day" "$1"
+    EMAIL_ADDRESS="$1"
+    TEMP_FILE=$(mktemp message.tmp.XXXXXXXXXXXXX)
+
+    cp "${RUN_DIR}/assets/template_head" "${TEMP_FILE}"
+    # add lucashifact
+    echo "${RANDOM_FACT_HTML}" >> "${TEMP_FILE}"
+    echo '</p>' >> "${TEMP_FILE}"
+
+    # add the image if necessary
+    if [ "$DICE_ROLL" -eq 1 ]; then
+        RANDOM_IMAGE=$(find "${RUN_DIR}/assets/img" -type f | shuf -n 1)
+        cat "${RUN_DIR}/assets/template_img" >> "${TEMP_FILE}"
+        base64 "${RANDOM_IMAGE}" >> "$TEMP_FILE"
+    else
+        echo '--body--' >> "${TEMP_FILE}"
+    fi
+    echo "--message--" >> "${TEMP_FILE}"
+
+    /usr/sbin/sendmail "${EMAIL_ADDRESS}" < "${TEMP_FILE}"
+    rm "${TEMP_FILE}"
 else
-    echo "$tmpstring"
+    echo "TODO"
```

Comment:
Can probably mimic the behaviour by building the tmp file no matter what, and simply echoing if not given an email